src/Communication/SAF.Communication.PubSub.Interfaces/ISubscription.cs

```diff
@@ -16,7 +16,7 @@ public interface ISubscription : IDisposable
 
         string[] Patterns { get; }
 
-        void With(Action<DateTimeOffset, Message> callback);
+        void SetHandler(Action<DateTimeOffset, Message> handler);
```

Comment:
We should check whether it is possible to avoid the `With` or `SetHandler` call completely by providing the handler action in the constructor of the implementing class?

Reply:
When we pass an action into the constructor we cannot decide whether its the Handler or the RawHandler.